### ISSUE TITLE
issue about FAD

### ISSUE THREAD

**Issue by zhenye234** [fcaspe/ddx7#2]
Hi, 
Thanks for the wonderful work.
I am also trying to replicate the Frechet Audio Distance score, and I also used Google's original FAD repo, but I achieved different results.
                                 Flute	Violin	Trumpet
Test Data	                1.3814	0.3003	0.7313
DDX7（I_max=2)	8.2608	7.0066	5.451

I delete this line in test() to get the ref_train.wav and calculate FAD between 
1, ref_train.wav and synth_test.wav
2, ref_train.wav and ref_test.wav

model = model.eval()
for k in self.loaders.keys():
    **if(k == 'train'): continue**
    print(k)

 
And I find that the FAD score I get is different from the results on the paper. So do you have any idea about the reason?

**Comment by fcaspe** [comment]
Hey zhenye234!
Thank you for your interest in my work!

About the results, one possible cause is that fornthe paper I computed the background distribution of all the files on the dataset, not only ref_train.wav. Could you recompute the background distribution and run FAD again?

Another thing is that FM could be quite difficult to optimize. I originally worked on a bigger repo and it could be that something changed when I cleaned up everything before uploading. I will check this.

Make sure that the FM configuration matches the instruments you are testing, and that you are using the same dataset train test split ratios.

I will upload the scripts I used to compute FAD, and will train some models again when I get some time. Let me know how it goes and if you want the pretrained models too.

**Comment by zhenye234** [comment]
Hi fcaspe,
I recompute the FAD using ref_train.wav, ref_test.wav, ref_val.wav together as background embedding and I find it can decrease about 0.2~0.3 of FAD. 

Also, I check the log and I find train loss descent well but val loss looks strange.  The order is violin, trumpet, flute
![1](https://user-images.githubusercontent.com/91582280/206625480-4a32ac0b-3181-401e-b6bb-8f204901234d.png)
![2](https://user-images.githubusercontent.com/91582280/206625499-52284b24-ad30-4ab5-8c5c-8b56882e4ad4.png)
![3](https://user-images.githubusercontent.com/91582280/206625509-cd6e2306-6d88-4a17-a625-441f381aeaae.png)
So do you have any idea about these loss curves?

And I really want the pre-trained models.
Thank you!
